ui/raidboss/data/07-dt/hunts/shaaloani.ts

```diff
@@ -434,6 +562,233 @@ const triggerSet: TriggerSet<Data> = {
+              inOut: output.outOfHitbox!(),
+              dir2: output[dir1]!(),
+              dir3: output[dir2]!(),
+            }),
+          };
+        }
+        return {
+          infoText: output.awayFrom!({ out: output.outOfHitbox!(), dir: awaySide }),
+        };
+      },
+    },
+    // The boss does either in; out; in-right + out-left; or out-right + in-left
+    // can be combined with sand orbs explosions.
+    {
+      id: 'Hunt Ttokrrone Desert Tempest',
+      type: 'StartsUsing',
+      netRegex: { id: ttokrroneDesertTempestIds, source: 'Ttokrrone', capture: true },
+      durationSeconds: 7,
+      alertText: (data, matches, output) => {
+        const tempest = ttokrroneDesertTempest[matches.id]!;
+        const pattern = data.ttokSandOrbPatterns[data.ttokSandOrbOnSet];
+        let inOut: TtokrroneDodge = 'unknown';
+        let dir2: TtokrroneDodge | null = null;
+        let dir3: TtokrroneDodge | null = null;
+        let backFrontSpot: TtokSafeSpots = 0b111_111_111_111;
+
+        if (tempest & TtokSafeSpots.In) {
+          inOut = 'in';
+        } else if (tempest & TtokSafeSpots.Out) {
+          inOut = 'out';
+        }
+        if (tempest === TtokSafeSpots.InRight) {
+          dir2 = 'right';
+        } else if (tempest === TtokSafeSpots.InLeft) {
+          dir2 = 'left';
+        }
+
+        if (pattern && data.ttokRotated === 0) {
+          const safeSpot = tempest & pattern;
+
+          if (safeSpot & TtokSafeSpots.Back) {
+            backFrontSpot = TtokSafeSpots.Back;
+            dir3 = 'back';
+          } else if (safeSpot & TtokSafeSpots.Front) {
+            backFrontSpot = TtokSafeSpots.Front;
+            dir3 = 'front';
+          }
+          // if the tempest was only in or out, but the sand pattern needs right or left
+          if (!dir2 && !(safeSpot & 0b010_010_010_010)) {
+            // we also need to check the other selected direction (front/back)
+            if (safeSpot & backFrontSpot & TtokSafeSpots.Right) {
+              dir2 = 'right';
+            } else if (safeSpot & backFrontSpot & TtokSafeSpots.Left) {
+              dir2 = 'left';
+            }
+          }
+          // swap in case dir2 is null, and its more natural phrasing
+          [dir2, dir3] = [dir3, dir2];
+          data.ttokSandOrbOnSet++;
+        }
+        if (dir2 && dir3) {
+          return output.triple!({
+            inOut: output[inOut]!(),
+            dir2: output[dir2]!(), // front or back
+            dir3: output[dir3]!(), // right or left
+          });
+        }
+        if (dir2) {
+          return output.double!({ inOut: output[inOut]!(), dir2: output[dir2]!() });
+        }
+        return output[inOut]!();
+      },
+      outputStrings: ttokrroneTempestSandspoutOutputStrings,
+    },
+    // Aoe
+    {
+      id: 'Hunt Ttokrrone Touchdown',
+      type: 'StartsUsing',
+      netRegex: { id: '91DB', source: 'Ttokrrone', capture: false },
+      durationSeconds: 4,
```

Comment:
```suggestion
```

No need to explicitly define `durationSeconds` for each trigger - it's treated as an override property.  Users can specify a default value for each warn level (`info`, `alert`, or `alarm`), and (I think?) the default-default is 4 seconds (maybe longer for `alarm`? I haven't looked in a while.).  Users can also override on a per-trigger basis in the config.

NB.  4-5 seconds is generally our rule of thumb anyway for how long we want to give users advanced warning to react to a mechanic, unless for some reason it's not possible.  We'll use longer durations for persistent mechanics (like 'stay out') or for things that require a little more conscious thought (like forced march), but that's very case-by-case.